src/mbgl/text/cross_tile_symbol_index.cpp

```diff
@@ -101,24 +129,41 @@ bool CrossTileSymbolLayerIndex::addBucket(const OverscaledTileID& tileID, Symbol
         }
     }
 
-    for (auto& symbolInstance: bucket.symbolInstances) {
-        symbolInstance.crossTileID = 0;
+    bucket.hasUninitializedSymbols = false;
+
+    if (tileID.overscaleFactor() > 1u) {
+        // For overscaled tiles the viewport might be showing only a small part of the tile,
+        // so we filter out the off-screen symbols to improve the performance.
+        for (auto& symbolInstance : bucket.symbolInstances) {
+            if (isInVewport(tileMatrix, symbolInstance.anchor.point)) {
+                symbolInstance.crossTileID = 0u;
+            } else {
+                symbolInstance.crossTileID = SymbolInstance::invalidCrossTileID();
```

Comment:
Would this delay symbol placement by 300ms for out-of-viewport symbols when we pan overscaled tile?

Reply:
it won't, as we reassign those ids at every frame